projects/plugins/jetpack/extensions/plugins/ai-assistant-plugin/components/breve/store/actions.ts

```diff
@@ -0,0 +1,74 @@
+/**
+ * Internal dependencies
+ */
+import { create, getTextContent, toHTMLString } from '@wordpress/rich-text';
+import features from '../features';
+import highlight from '../highlight/highlight';
+
+export function setPopoverState( isOpen ) {
+	return {
+		type: 'SET_POPOVER_STATE',
+		isOpen,
+	};
+}
+
+export function setBlockText( block ) {
+	return {
+		type: 'SET_BLOCK_TEXT',
+		clientId: block.clientId,
+		text: block.text,
+		index: block.index,
+	};
+}
+
+export function setBlockContent( clientId ) {
+	return ( { registry: { select, dispatch: dispatchFromRegistry }, dispatch } ) => {
+		const block = select( 'core/block-editor' ).getBlock( clientId );
+		const blocks = select( 'jetpack/ai-breve' ).getBlocksContent();
+		const blockIndex = blocks.findIndex( b => b.clientId === clientId );
+		const savedText = blocks[ blockIndex ]?.text ?? '';
+
+		const currentContent =
+			typeof block.attributes.content === 'string'
+				? create( { html: block.attributes.content } )
+				: create( block.attributes.content );
```

Comment:
Blocks don't have to set their text content on `attributes.content`. Tables do not and the complex words are not detected there.

For the readability calculation we are using `getBlockContent` to get the block's HTML and then stripping the tags out. For that case it works, but here we'll need some more thought :thinking: